cli/src/pcluster/config/cluster_config.py

```diff
@@ -2284,16 +2403,17 @@ def _register_validators(self):
                 checked_images.append(queue_image)
                 self._register_validator(AmiOsCompatibleValidator, os=self.image.os, image_id=queue_image)
             for compute_resource in queue.compute_resources:
-                self._register_validator(
-                    InstanceTypeBaseAMICompatibleValidator,
-                    instance_type=compute_resource.instance_type,
-                    image=queue_image,
-                )
-                self._register_validator(
-                    InstanceArchitectureCompatibilityValidator,
-                    instance_type=compute_resource.instance_type,
-                    architecture=self.head_node.architecture,
-                )
+                for instance_type in compute_resource.instance_types:
+                    self._register_validator(
+                        InstanceTypeBaseAMICompatibleValidator,
+                        instance_type=instance_type,
+                        image=queue_image,
+                    )
+                    self._register_validator(
+                        InstanceArchitectureCompatibilityValidator,
+                        instance_type=instance_type,
```

Comment:
let's think about passing directly a list of InstanceTypeInfo and we move the architecture method in there